### PR TITLE
feat(config): allow user to provide a path to funding.yml

### LINKED ISSUE #389: Funding is not detected
With [this](https://github.com/MarcoIeni/release-plz/pull/826/commits/f8ff146bd00d969d69a16287f1d46cea37ac7d0e) configuration, oranda doesn't detect funding.
image

I ran `oranda build && oranda serve` from the `/crates/release_plz` directory.
I tried running oranda from the root directory of the project specifying `--project-root` and/or `--config-path` but none of them worked. 
I also tried adding `"funding": true,` to the config file.

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #389

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
this doesn't work yet but i do not know yet why

File: src/config/oranda_config/funding.rs [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+        self.yml_path.apply_opt(layer.yml_path);
+        self.md_path.apply_opt(layer.md_path);
+    }
+}
+
+impl FundingConfig {
+    // see if we can find a FUNDING.yml, if so update the config path
+
+    /// If we have a FUNDING.yml file, try to find it. If we fail, we disable funding support.
+    pub fn find_paths(real_this: &mut Option<Self>) -> Result<()> {
```
Comment (ashleygwilliams) [inline src/config/oranda_config/funding.rs]: `real_this` is not immediately understandable as a name for me- maybe we can say UserConfig or something more descriptive about the origin of the data?

File: src/config/oranda_config/funding.rs [hunk 2]
```diff
@@ -1,17 +1,64 @@
...
+        let Some(this) = real_this else {
```
Comment (ashleygwilliams) [inline src/config/oranda_config/funding.rs]: i did not actually realize you could do this

File: src/config/oranda_config/funding.rs [hunk 3]
```diff
@@ -1,17 +1,64 @@
...
+        }
+        // Try to auto-detect funding.md if not specified
+        if this.md_path.is_none() {
+            let default_md_path = Utf8PathBuf::from("./funding.md");
+            if default_md_path.exists() {
+                this.md_path = Some(default_md_path.to_string());
+            }
+        }
+
+        let missing_important_things = this.yml_path.is_none() && this.md_path.is_none();
```
Comment (ashleygwilliams) [inline src/config/oranda_config/funding.rs]: maybe `cant_find_files` ?

File: src/config/oranda_config/funding.rs [hunk 4]
```diff
@@ -1,17 +1,64 @@
...
+        let customized_other_things = this.preferred_funding.is_some();
```
Comment (ashleygwilliams) [inline src/config/oranda_config/funding.rs]: `has_user_config` ?

**Review by ashleygwilliams — COMMENTED** [review]
i can't approve because in theory this is my PR, but i would say modulo perhaps some naming updates this looks good

**Review by Gankra — APPROVED** [review]

**Review by manyinsects — APPROVED** [review]

**Pull request merged**